src/api/things.ts

```diff
@@ -0,0 +1,31 @@
+import { z } from "zod"
+
+import { withRouteSpec } from "lib/middleware/index.ts"
+import * as m from "lib/models.ts"
+
+const jsonBody = m.thingInitializer
+
+const jsonResponse = z.object({
+  thing: m.thing.optional(),
+  things: z.array(m.thing).optional(),
+})
+
+export default withRouteSpec({
+  auth: "api_key",
+  methods: ["GET", "POST"],
+  jsonBody: jsonBody.optional(),
```

Comment:
Shouldn't it be possible to not have this be optional w/ `shouldValidateGetRequestBody: false`?

Reply:
Or I guess we are removing `shouldValidateGetRequestBody` with method discrimination?

Reply:
personally I think we should remove `shouldValidateGetRequestBody` and prefer the `.optional()` style
in the HTTP spec it's invalid to provide a body for GET so I think we can probably just ignore it